tsdb/index/postings.go

```diff
@@ -224,21 +224,27 @@ func (p *MemPostings) All() Postings {
 
 // EnsureOrder ensures that all postings lists are sorted. After it returns all further
 // calls to add and addFor will insert new IDs in a sorted manner.
-func (p *MemPostings) EnsureOrder() {
+// Parameter numberOfConcurrentProcesses is used to specify the maximal number of
+// CPU cores used for this operation. If it is <= 0, GOMAXPROCS(0) is used.
+// GOMAXPROCS(0) was the default before introducing this parameter.
+func (p *MemPostings) EnsureOrder(numberOfConcurrentProcesses int) {
```

Comment:
Hi @pracucci, 
With @codesome we discussed that the signature of `EnsureOrder` should be changed by adding a parameter representing the number of CPU cores used for this operation. If this value is <= 0, then `GOMAXPROCS` should be used.
We haven't explicitly discussed about the implementation, but since `EnsureOrder` is an exported method, the argument that is passed here could have any value. Currently, the only usage of `EnsureOrder` in prometheus is in `Head.Init()`, where `h.opts.MemPostingsEnsureOrderConcurrency` is passed as argument. I was thinking to add a check similar to the one done for `WalReplayConcurrency` and to set its default value to be `GOMAXPROCS`, but then we would have the same check in 2 different places: one in `Head.NewHead()`, that would validate the flag, and the other in `EnsureOrder()`, that would validate the passed argument. This was also my initial implementation, but then I decided to remove it in order to avoid 2 similar checks. I can easily add this check again.

Reply:
Hmm, I think the default should be GOMAXPROCS since that was the default before this change. So the consumer of this method has to just pass `0` if they want to keep the concurrency the same after this change. We are not introducing a `MemPostingsEnsureOrderConcurrency`, and re-using `WALReplayConcurrency`, which already has this check and sets default to GOMAXPROCS.